src/hypertrace/agent/config/__init__.py

```diff
@@ -110,169 +103,227 @@ def __init__(self):  # pylint: disable=R0912,R0915
 
         if 'HT_REPORTING_TOKEN' in os.environ:
             logger.debug("[env] Loaded HT_REPORTING_TOKEN from env")
-            reporting_token = os.environ['HT_REPORTING_TOKEN']
+            self.config['reporting']['token'] = os.environ['HT_REPORTING_TOKEN']
```

Comment:
Shall we check `self.config['reporting']` exist?

Reply:
Fixed.